app/models/user.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+require 'byebug'
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:facebook]
+
+  validates :first_name, presence: true
+  after_create :create_profile
+ 
+
+
+  has_one :profile, dependent: :destroy
+  accepts_nested_attributes_for :profile
+
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+
+  has_many :comments, foreign_key: :commenter_id, dependent: :destroy
+
+  has_many :sent_requests, class_name: 'FriendRequest', foreign_key: :sender_id, dependent: :destroy
+  has_many :friends_requested, through: :sent_requests, source: :receiver
+
+  has_many :received_requests, class_name: 'FriendRequest', foreign_key: :receiver_id, dependent: :destroy
+  has_many :friends_requesting, through: :received_requests, source: :sender
+
+  has_many :direct_friendships, class_name: 'Friendship', foreign_key: :user_id, dependent: :destroy
+  has_many :direct_friends, through: :direct_friendships, source: :friend
+
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: :friend_id, dependent: :destroy
+  has_many :inverse_friends, through: :inverse_friendships, source: :user
+
+  has_many :likes, foreign_key: :liker_id, dependent: :destroy
+
+  has_many :liked_posts, through: :likes, source: :likeable, source_type: 'Post'
+  has_many :liked_comments, through: :likes, source: :likeable, source_type: 'Comment'
+
+  def friends
+    direct_friends + inverse_friends
+  end
+
+  def friend?(user_id)
+    friends.map(&:id).include?(user_id)
+  end
+
+  def friend_requested?(user_id)
+    friends_requested.pluck(:id).include?(user_id)
+  end
+
+  def friend_requesting?(user_id)
+    friends_requesting.pluck(:id).include?(user_id)
+  end
+
+  def send_friend_request(user_id)
+    sent_requests.create(receiver_id: user_id)
+  end
+
+  def delete_friend_request(user_id)
+    received_requests.where(sender_id: user_id).first.destroy
+  end
+
+  def add_friend(user_id)
```

Comment:
In creating `Friendship` records we probably don't need this additional special instance method in `user`. Simple `create` action in a `friendships_controller` would be better.